### DISCUSSION TITLE
Problems Pane referencing file that doesn't exist

### DISCUSSION THREAD

**Discussion by brndngrhm** [posit-dev/positron#4482] [Q&A]
I noticed i had a lot of problems in the problems pane, but they all seem to be referencing some file called `.vdoc.r`, which doesn't seem to exist - see screnshot below which is what opened after i double clicked the error, maybe it's some sort of temporary file? Is there a way to maybe ignore `.vdoc` files?

![image](https://github.com/user-attachments/assets/958f112b-af9e-4e40-ba9f-8a6ea57009b4)

**Comment by juliasilge** [comment]
Oh interesting 👀 

That is coming from the Quarto extension, and it is required for Quarto to know how to handle the R code in your file. Here is the code that makes it:

https://github.com/quarto-dev/quarto/blob/843625f46ae3efe77598d50266791b63f7c1867f/apps/vscode/src/vdoc/vdoc-tempfile.ts#L45

Do you have a reproducible set of steps that generate these problems for you, with a `.qmd` that you can share? If you can create that, I recommend sharing it on the Quarto extension repo: https://github.com/quarto-dev/quarto/issues

I also am curious what is going on, so do cc me if you can create a reproducible example!

**Reply by baderstine** [reply]
same issue here.  I have 111 problems listed and they all appear to be related to `.vdoc.(random string of characters).r` files that I cannot find. I am editing a .Rmd file and would like to be able to see only Problems that have to do with the file i am currently editing.  e.g. `[{
	"resource": "/var/folders/c6/y9gkwb0n7cb07dlwf462zb9c0000gq/T/tmp-30485-PDlNetrSZMYW/.vdoc.2e1016cb-fadb-4a78-a20e-b64edd1cc662.r",
	"owner": "_generated_diagnostic_collection_name_#3",
	"severity": 8,
	"message": "Unmatched opening delimiter. Missing a closing ')'.",
	"startLineNumber": 176,
	"startColumn": 41,
	"endLineNumber": 176,
	"endColumn": 42,
	"origin": "extHost1"
}]` 

If I click on an error I get the message in the editor window: 
`The editor could not be opened because the file was not found.` 

So seems like old Problems for files that no longer exist are sticking around cluttering up the Problems pane. There does not appear to be a way to remove these.

**Reply by baderstine** [reply]
The problems seem to stack up on one another every 10 seconds or so. Perhaps it is related to autosave files being generated in the background but not being cleared? If I have a file open with 2 errors, initially there are no problems, then after 10 (or so) seconds there are 2 errors in temp file 1, and then a few seconds later, 4 errors in temp file 1+2, then 6, then 8, etc. If I fix the errors, eventually they begin to delete themselves, but seems like a bug where the Problems list should only have a single file listed (the one that I'm editing), rather than an ever increasing/decreasing list of temp files?

**Reply by juliasilge** [reply]
Can you check what version of the Quarto extension (not the Quarto CLI) you have? I ask because we did some recent work in https://github.com/quarto-dev/quarto/pull/832 to address https://github.com/quarto-dev/quarto/issues/819, and my understanding is that this should be fixed in the current release of the Quarto extension.

**Reply by baderstine** [reply]
Looks like quarto.quarto v1.126.0

**Reply by baderstine** [reply]
It may be specific to .Rmd files and sounds exactly the same as quarto-dev/quarto#819.  When I'm editing a .R file I do not experience the same issue.  I have tried restarting Positron multiple times.

**Reply by juliasilge** [reply]
That is the updated version of the extension, yep.

Can you share a set of steps that causes the problems/diagnostics to be reported for you? The original report was for diagnostics that would be generated when you save a `.qmd` or `.Rmd` file with a syntax error in it. Right now, if I have a `.Rmd` and add a syntax error (like say `rnorm(10. 1)`), I do not see any diagnostics, even after saving.

**Reply by baderstine** [reply]
Similar to what you've described.  Create a new `.Rmd` file and create a cell with a syntax error.  e.g., 
```
---
title: "title"
author: "author"
date: "`r Sys.Date()`"
output: 
  html_document: 
    toc: yes
    toc_float: true
---

\```{r}
r norm(1, 1)
\```
```

When I save this document, it is automatically formatted as: 
```
r
norm(1,1) 
``` 

and I get a warning in the Problems list: 
```
[{
	"resource": "<pathto>/.vdoc.5feafe64-6614-4f48-bc8a-bde2a9b4a2fd.r",
	"owner": "_generated_diagnostic_collection_name_#3",
	"severity": 4,
	"message": "No symbol named 'r' in scope.",
	"startLineNumber": 12,
	"startColumn": 1,
	"endLineNumber": 12,
	"endColumn": 2,
	"origin": "extHost1"
}]
```

**Reply by juliasilge** [reply]
Ah, I think it is the formatting that is involved here! I opened https://github.com/quarto-dev/quarto/issues/855 to track and prioritize a fix.

As a temporary workaround, you might turn off format-on-save for Quarto and use _Format Document_ from the command palette at points when you don't think you have syntax errors.

**Reply by baderstine** [reply]
ok that makes sense.
